src/vm/threads.h

```diff
@@ -7039,6 +7010,8 @@ struct ThreadLocalInfo
     Thread* m_pThread;
     AppDomain* m_pAppDomain; // This field is read only by the SOS plugin to get the AppDomain
     void** m_EETlsData; // ClrTlsInfo::data
+    EventPipeBufferList* m_pThreadEventBufferList;
```

Comment:
@jkotas @kouvel - Is there rationale about why some fields are placed directly in ThreadLocalInfo vs. most others with are indirected inside m_EETlsData? Its unclear to me if putting new fields here is appropriate.

Reply:
I think you should be able do define your own local `__thread EventPipeBufferList* t_pThreadEventBufferList`. The reasons why we had to have all thread-local statics in this central structure should no longer exist.

Reply:
> walk me through the rationale of why they are no longer necessary.

The central structure was required because of the offsets of thread-local statics were hardcoded in the debugger. They are no longer hardcoded. The code in `InitThreadManager` computes the current offset and communicates it to debugger via DAC.

Reply:
@jkotas - sorry I had two separate issues in those two comments. The text you quoted was in the first comment asking about Sung changing types from Volatile<T> to T but you answered referring to the second comment about usage of TLS slots.

Reply:
@noahfalk Yeah, I meant to put the Volatile there. Thanks for catching that!
@jkotas Regarding where these are placed (in ThreadLocalInfo vs just defining ```__thread EventPipeBufferList* t_pThreadEventBufferList``` - is one preferred over the other?

Reply:
My preference would be the local definition - `__thread EventPipeBufferList* t_pThreadEventBufferList`.